pages/deliverer/incoming-orders/[id].tsx

```diff
@@ -104,30 +86,38 @@ const IncomingOrder = ({ user }) => {
     } else {
       setAlert({ type: 'info', message: 'Rejected order' })
     }
-    router.push('/home')
+    router.push('/deliverer/home')
   }
 
   async function handleAccept() {
-    // implement auth to get the user uid
     const { data: updatedOrdersData, error: updatedOrdersError } =
       await supabaseClient
         .from('orders')
-        .update({ id: id, deliverer_id: user.id })
+        .update({ id: id, deliverer_id: user.id, accepted_at: new Date() })
         .eq('id', id)
         .is('is_active', true)
         .is('deliverer_id', null)
 
-    // possible that order has been taken or completed
-    if (updatedOrdersError) {
+    // remove order from pending_orders array in users table
+    const { error: removePendingError } = await supabaseClient.rpc(
+      'remove_pending_order', 
+      { order_id: id, user_id: user.id }
+    )
```

Comment:
shifted this to before the checks for error, so that we remove the pending order before we push to home